### DISCUSSION TITLE
How do I make providerAccountId accessible on session object

### DISCUSSION THREAD

**Discussion by satoufuyuki** [nextauthjs/next-auth#1486] [Help] [answered]
I want `providerAccountId` accessible on session object. I know there is callback function to extend the properties. But I couldn't find any `providerAccountId` there.

**Comment by StefanSelfTaught** [comment, marked as answer]
I think you can get your `providerAccountId` in the  `profile` or  `account` object in the JWT callback and then pass it to the session like this:

```js
callback: {
  async session(session, token) {
      session.providerAccountId = token.providerAccountId 
      return session
  },
  async jwt(token, user, account, profile, isNewUser) {
    if (profile?.id) {
        token.providerAccountId = profile.id
    }
      return token
  }
}

**Reply by satoufuyuki** [reply]
it works! Thanks
